packages/pass-style/src/passStyleOf.js

```diff
@@ -81,28 +81,28 @@ const makePassStyleOf = passStyleHelpers => {
    * structures, so without this cache, these algorithms could be
    * O(N**2) or worse.
    *
-   * @type {WeakMap<Passable, PassStyle>}
+   * @type {WeakMap<WeakKey, PassStyle>}
    */
   const passStyleMemo = new WeakMap();
 
   /**
    * @type {PassStyleOf}
    */
+  // @ts-expect-error cast
   const passStyleOf = passable => {
     // Even when a WeakSet is correct, when the set has a shorter lifetime
     // than its keys, we prefer a Set due to expected implementation
     // tradeoffs.
     const inProgress = new Set();
 
-    /**
-     * @type {PassStyleOf}
-     */
     const passStyleOfRecur = inner => {
       const innerIsObject = isObject(inner);
       if (innerIsObject) {
-        if (passStyleMemo.has(inner)) {
-          // @ts-ignore TypeScript doesn't know that `get` after `has` is safe
-          return passStyleMemo.get(inner);
+        const innerMemo = passStyleMemo.get(inner);
+        if (innerMemo) {
+          // sanity check on falsy values. Do 'get' first to get the type narrowing of the `if`
+          assert(passStyleMemo.has(inner));
```

Comment:
What value is this sanity check providing, given that `get` is guaranteed to return undefined for absent keys?
```suggestion
```

Reply:
indeed

Reply:
I don't think so, because `passStyleMemo` values are always PassStyle strings. And if we do care, I guess the place for such an assertion would be adjacent to `set` rather than `get`.

Reply:
Unfortunately, simpy knowledge that it is a string does not by itself imply that it is truthy, since the empty string is falsy. In this case, we know that it is always a pass-style string, which is always a non-empty string and thus always truthy. But still, to avoid even needing to worry about this edge case, I suggest rewriting

```js
if (inner) {
```
to
```js
if (inner !== undefined) {
```
or
```js
if (typeof inner === 'string') {
```
FWIW, I prefer the `!== undefined` .

Reply:
In any case, given that we statically know we're only putting strings in there, I'd say no further sanity check is needed.